rockcraft/lifecycle.py

```diff
@@ -36,23 +36,51 @@ def pack():
     except OSError as err:
         raise CraftError(err) from err
 
+    name = yaml_data.get("name")
+    if not name:
+        raise CraftError("Project name not specified.")
+
+    tag = yaml_data.get("tag")
+    if not name:
+        raise CraftError("Project tag not specified.")
+
+    base = yaml_data.get("base")
+    if not base:
+        raise CraftError("Base not specified.")
+
     parts_data = yaml_data.get("parts")
     if not parts_data:
-        emit.message("No parts specified.")
-        return
+        raise CraftError("No parts specified.")
+
+    work_dir = Path("build").absolute()
```

Comment:
no change for now, but calling this build can be problematic as build is also a step for a part and has a directory. I cannot think of a better name?